File: services/rag/src/chunking/tokenizer.ts
import { getEncoding } from "js-tiktoken";

// cl100k_base is used by text-embedding-3-* and GPT-4 models
const enc = getEncoding("cl100k_base");

/**
 * Count tokens in a string using the cl100k_base BPE tokenizer.
 * More accurate than the 1-token-per-4-chars heuristic, especially
 * for code, URLs, and non-English text.
 */
export function countTokens(text: string): number {
  return enc.encode(text).length;
}

File: services/confluence-importer/src/confluence_importer/chunking/tokenizer.py
"""Token counting with the cl100k_base BPE tokenizer (used by text-embedding-3-*)."""

from functools import lru_cache

import tiktoken


@lru_cache(maxsize=1)
def get_encoding() -> tiktoken.Encoding:
    return tiktoken.get_encoding("cl100k_base")


def count_tokens(text: str) -> int:
    """More accurate than the 1-token-per-4-chars heuristic, especially for code and URLs."""
    return len(get_encoding().encode(text))
